CMakeLists.txt

```diff
@@ -107,43 +126,26 @@ if (CODE_COVERAGE)
 endif()
 
 ###############################################################################
-# Compile application
+# Default shared libraries
 ###############################################################################
+# Global flag to cause add_library() to create shared libraries if on.
+# If set to true, this will cause all libraries to be built shared
+# unless the library was explicitly added as a static library.
+option(BUILD_SHARED_LIBS "Create shared libraries by default" ON)
 
-# Project sources
-file(GLOB_RECURSE SOURCES_FILES
-        "src/**/*.c"
-        "src/**/*.cpp"
-        "src/**/*.cxx"
-    )
-
-# Project headers
-file(GLOB_RECURSE HEADERS_FILES
-        "include/**/*.h"
-        "include/**/*.hpp"
-        "include/**/*.hxx"
-    )
-
-add_executable(${PROJECT_NAME}
-        ${SOURCES_FILES}
-        ${HEADERS_FILES}
-    )
+###############################################################################
+# Compile library.
+###############################################################################
+add_subdirectory(src/cpp)
 
-target_include_directories(${PROJECT_NAME} PRIVATE
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/types>
-    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
-    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include/${PROJECT_NAME}>)
-
-target_link_libraries(${PROJECT_NAME}
-    fastrtps
-    fastcdr
-    yaml-cpp
-    $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
-    )
+###############################################################################
+# Compile application
+###############################################################################
+option(COMPILE_TOOLS "Compile and install tools" ON)
```

Comment:
Be careful, tool tests should not be active if COMPILE_TOOLS  is OFF.

Reply:
Good catch :+1: